code/game/objects/items/stacks/medical.dm

```diff
@@ -29,31 +31,34 @@
 		user << "<span class='danger'>You don't have the dexterity to do this!</span>"
 		return 1
 
-	if (user)
+	if(isliving(M))
+		if(!M.can_inject(user, 1))
+			return
+
+	if(user)
 		if (M != user)
-			user.visible_message( \
-				"<span class='notice'>[user] has applied [src] on [M].</span>", \
-				"<span class='notice'>You apply \the [src] to [M].</span>" \
-			)
+			user.visible_message("<span class='green'>[user] applies [src] on [M].</span>", "<span class='green'>You apply [src] on [M].</span>")
 		else
 			var/t_himself = "itself"
 			if (user.gender == MALE)
 				t_himself = "himself"
 			else if (user.gender == FEMALE)
 				t_himself = "herself"
+			user.visible_message("<span class='notice'>[user] starts to apply [src] on [t_himself]...</span>", "<span class='notice'>You begin applying [src] on yourself...</span>")
+			if(!do_mob(user, M, self_delay))	return
+			user.visible_message("<span class='green'>[user] applies [src] on [t_himself].</span>", "<span class='green'>You apply [src] on yourself.</span>")
 
-			user.visible_message( \
-				"<span class='notice'>[M] applied [src] on [t_himself].</span>", \
-				"<span class='notice'>You apply \the [src] on yourself.</span>" \
-			)
-
-	if (istype(M, /mob/living/carbon/human))
 
+	if(istype(M, /mob/living/carbon/human))
 		var/mob/living/carbon/human/H = M
 		var/obj/item/organ/limb/affecting = H.get_organ(check_zone(user.zone_sel.selecting))
 
+		if(stop_bleeding)
+			if(!H.bleedsuppress) //so you can't stack bleed suppression
+				H.suppress_bloodloss(stop_bleeding)

```

Comment:
But you can also pre-emptively use gauzes so you won't ever need to worry about bleeding. Each gauze protects for five minutes, so just carry few with you, use one when last one wears off, and you'll never bleed.